spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/support/AbstractRibbonCommand.java

```diff
@@ -74,9 +84,11 @@ protected static Setter getSetter(final String commandKey) {
 	@Override
 	protected ClientHttpResponse run() throws Exception {
 		final RequestContext context = RequestContext.getCurrentContext();
-		String contentLengthHeader = context.getRequest().getHeader("Content-Length");
-		if (StringUtils.hasText(contentLengthHeader)) {
-			this.context.setContentLength(new Long(contentLengthHeader));
+		if (context.getRequest() != null) {
```

Comment:
When you use thread isolation, RequestContext is empty because it is stored in the ThreadLocal. It works for Semaphore isolation as RibbonCommand is handled in the same thread.